modules/tensor_mechanics/src/kernels/ALEKernel.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ALEKernel.h"
+
+template<>
+InputParameters validParams<ALEKernel>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Sets up derivetives with respect to initial configuration");
+  return params;
+}
+
+ALEKernel::ALEKernel(const InputParameters & parameters) :
+  Kernel(parameters),
+  _assembly_undisplaced(_fe_problem.assembly(_tid)),
+  _grad_phi_undisplaced(_assembly_undisplaced.gradPhi())
+{
+}
+
+void
+ALEKernel::computeJacobian()
+{
+  _fe_problem.prepareShapes(_var.number(), _tid);
```

Comment:
Ok - this is required... but I don't like it.  You can see what happens here:

https://github.com/idaholab/moose/blob/devel/framework/src/base/ComputeJacobianThread.C#L65

That line means that only the variable this Kernel is operating on (which will be the displaced version of that variable) gets its shapes prepared... so this is necessary.